Add Entry: General: Serialization (#634)
Codecademy/docs: content/general/concepts/serialization-deserialization/serialization-deserialization.md

---
Title: 'Serialization/Deserialization'

> Description: 'Serialization is the process of translating a complex object into a format that can be later deserialized back into a copy of the original object.'

Reviewer: ```suggestion
Description: 'Serialization is the process of translating a complex object into a format that can be deserialized back into a copy of the original object.'
```

Subjects:
  - 'Computer Science'
  - 'Code Foundations'
Tags:
  - 'Classes'
  - 'Data Structures'
  - 'JSON'
  - 'Objects'
CatalogContent:
  - 'learn-java'
  - 'paths/computer-science'
---

> Serialization is the process of translating a complex object into a format, typically a string of characters, that can be later deserialized back into a copy of the original object. This is beneficial as it allows the software to save the state of an arbitrary object to a destination that can be completely unaware of the details of the object being saved. So any object can be saved to a simple string variable, a database field, an external file system, some third-party application in the cloud, or even printed out on a sheet of paper.

Reviewer: ```suggestion
Serialization is the process of translating a complex object into a format, typically a string of characters, that can be deserialized back into a copy of the original object. This is beneficial as it allows the software to save the state of an arbitrary object to a destination that can be completely unaware of its details. Any object can be saved to a simple string variable, database field, external file system, third-party application in the cloud, or even printed out on a sheet of paper.
```

> Common formats for serialization include XML and JSON, which have the advantage of being human-readable, but there are also binary formats that aren't human-readable. Regardless of the format, the purpose of serialization is that at a later point in time the serialized data can be read or processed as input and be deserialized back into a copy of the original object.

Reviewer: ```suggestion
Common formats for serialization include XML and JSON, which have the advantage of being human-readable. However, other formats like binary aren't human-readable. Regardless of the format, the purpose of serialization is that at a later point in time, the serialized data can be read or processed as input and be deserialized back into a copy of the original object.
```
